metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/WorkflowUtils.java

```diff
@@ -262,7 +262,9 @@ public static AbstractExecutablePlugin computePredecessorPlugin(ExecutablePlugin
 
     // If the plugin type does not need a predecessor we are done.
     final Set<ExecutablePluginType> predecessorTypes = getPredecessorTypes(pluginType);
-    if (predecessorTypes.isEmpty()) {
+    if (predecessorTypes.isEmpty()
+        || pluginType == ExecutablePluginType.DEPUBLISH && predecessorTypes.size() == 1
+        && predecessorTypes.contains(ExecutablePluginType.PUBLISH)) {
```

Comment:
Is the precedence of operations clear here? Is it && before || ? Or is it perhaps better to add parentheses?

Reply:
The result wouldn't change, but I added them for clarity